src/hotspot/share/opto/loopnode.cpp

```diff
@@ -3907,6 +3907,33 @@ bool PhaseIdealLoop::is_deleteable_safept(Node* sfpt) {
 
 //---------------------------replace_parallel_iv-------------------------------
 // Replace parallel induction variable (parallel to trip counter)
+// This optimization looks for patterns similar to:
+//
+//    int a = init2;
+//    for (int i = init; i < limit; i += stride) {
+//      a += stride2;
+//    }
+//
+// and transforms it to:
+//
+//    int a = init2
+//    for (int i = init; i < limit; i += stride) {
+//      a = init2 + (i - init) * (stride2 / stride)
```

Comment:
I like that you are putting comments here, I think it will help. But they seem not quite correct.
If `stride2 = 2` and `stride = 4`, then the division would be rounded down to zero.
Can you be more precise about the order of operations, and rounding issues?

Reply:
Can you also be consistent with the names all the way through your comments? I suggest you just only use `stride_con`, and not `stride`. You can use `i` and `a`, if you want. But then it would be helpful if you had two lines with identical expressions, but where you make the transition from `i` to `phi`.

Reply:
Ah. It seems that we require `stride2 / stride` to be a lossless division in the code. A comment about that limitation would be helpful. And I think you should also check if there are tests that cover cases where the division would be lossy.

Reply:
> ...be consistent with the names...

I had this concern, especially `i` vs `phi`. I didn't think it was reasonable to call the iterator `phi` only because the optimization code calls such a value so by extracting from the phi node. I agree to keep things consistent. The example to trivial to be understood anyway. I updated the naming.

> It seems that we require stride2 / stride to be a lossless division in the code.

Not only lossless division (i.e., rounding-towards-zero) is used, the optimization requires this division to be exact with no remainders. Checks are in place to make sure optimization only happens if this condition is met.

I updated the comments to be more expressive regarding this.